controllers/fenceagentsremediation_controller_test.go

```diff
@@ -0,0 +1,175 @@
+			When("FAR's name isn't a node name", func() {
+				It("should fail", func() {
+					underTestFAR.ObjectMeta.Name = dummyNodeName
+					_, err := buildFenceAgentParams(underTestFAR)
+					Expect(err).To(HaveOccurred())
+				})
+				It("should succeed", func() {
+					underTestFAR.ObjectMeta.Name = validNodeName
+					_, err := buildFenceAgentParams(underTestFAR)
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+		})
+		When("creating a resource", func() {
+			It("should fail when FAR pod is missing", func() {
+				//Test getFenceAgentsPod func
+			})
+		})
+	})
+
+	Context("Reconcile", func() {
+		//Scenarios
+
+		BeforeEach(func() {
+			// Create fenceAgentsPod and FAR
+			Expect(k8sClient.Create(context.Background(), fenceAgentsPod)).NotTo(HaveOccurred())
+			Expect(k8sClient.Create(context.Background(), underTestFAR)).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			Expect(k8sClient.Delete(context.Background(), fenceAgentsPod)).NotTo(HaveOccurred())
+			Expect(k8sClient.Delete(context.Background(), underTestFAR)).NotTo(HaveOccurred())
+		})
+
+		When("creating FAR CR", func() {
+			It("should build the exec command based on FAR CR", func() {
+				Eventually(func() (bool, error) {
+					res, err := cliCommandsEquality(underTestFAR)
+					return res, err
+				}, 1*time.Second, 500*time.Millisecond).Should(BeTrue(), BeEmpty())
+			})
+		})
+	})
+})
+
+// newFenceAgentsRemediation assign the input to the FenceAgentsRemediation's Spec
+func newFenceAgentsRemediation(nodeName string, agent string, sharedparameters map[v1alpha1.ParameterName]string, nodeparameters map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string) *v1alpha1.FenceAgentsRemediation {
+	return &v1alpha1.FenceAgentsRemediation{
+		ObjectMeta: metav1.ObjectMeta{Name: nodeName, Namespace: defaultNamespace},
+		Spec: v1alpha1.FenceAgentsRemediationSpec{
+			Agent:            agent,
+			SharedParameters: sharedparameters,
+			NodeParameters:   nodeparameters,
+		},
+	}
+}
+
+// buildFarPod builds a dummy pod with FAR label and namespace
+func buildFarPod() *corev1.Pod {
+	fenceAgentsPod := &corev1.Pod{}
+	fenceAgentsPod.Labels = faPodLabels
+	fenceAgentsPod.Name = "mock-fence-agents"
+	fenceAgentsPod.Namespace = defaultNamespace
+	container := corev1.Container{
+		Name:  "foo",
+		Image: "foo",
+	}
+	fenceAgentsPod.Spec.Containers = []corev1.Container{container}
+	return fenceAgentsPod
+}
+
+// cliCommandsEquality creates the command for CLI and compares it with the production command
+func cliCommandsEquality(far *v1alpha1.FenceAgentsRemediation) (bool, error) {
+	//fence_ipmilan --ip=192.168.111.1 --ipport=6233 --username=admin --password=password --action=status --lanplus
+	if mocksExecuter.command == nil {
+		return false, errors.New("The command from mocksExecuter is null")
+	}
+
+	// hardcode expected command
+	expectedCommand := []string{" ", "--lanplus", "--password=password", "--username=admin", "--action=reboot", "--ip=192.168.111.1", "--ipport=6233"}
```

Comment:
one last comment 😉 
would it make sense to set a better agent in line 65 than `" "` and test for it here? 🙂

Reply:
It wouldn't hurt 😄